synapse/config/workers.py

```diff
@@ -50,13 +51,20 @@ def _instance_to_list_converter(obj: Union[str, List[str]]) -> List[str]:
     return obj
 
 
-@attr.s(auto_attribs=True)
-class InstanceLocationConfig:
+class InstanceLocationConfig(BaseModel):
```

Comment:
Discussion: by default, pydantic will ignore unknown fields (see `extra` under https://docs.pydantic.dev/usage/model_config/). We explicitly opted in to that behaviour for parsing request bodies:

https://github.com/matrix-org/synapse/blob/d642ce4b3258012da6c024b0b5d1396d2a3e69dd/synapse/rest/models.py#L4-L23

(though in retrospect, that seems unnecessary.)

Do we want to ignore extra fields here too, or explicitly reject them?

It might also be nice to include `allow_mutation=False` here (though the old attrs model didn't have  frozen=True, so this isn't needed for a like-to-like replacement.)

Reply:
I'm inclined to just duplicate this. I believe ignoring extra fields is fairly safe and disallowing changing the data is a good idea to prevent accidents. Do you think bringing all the comments is appropriate, or should I just include a comment pointing at `RequestBodyModel` for a lot of the why? My instinct is just to include them all as they appear relevant.

Reply:
> Do you think bringing all the comments is appropriate, or should I just include a comment pointing at RequestBodyModel for a lot of the why? 

No strong opinions either way.

> I believe ignoring extra fields is fairly safe

Nothing would go wrong, but it might be worth making admins aware that they've written some config that doesn't do anything (by rejecting unknown fields). Probably go with a direct copy for now and see how the approach fairs in  practice.

Reply:
> Nothing would go wrong, but it might be worth making admins aware that they've written some config that doesn't do anything (by rejecting unknown fields). Probably go with a direct copy for now and see how the approach fairs in practice.

I played around with 'forbid'ding extra fields, and while I like the functionality the errors it raises weren't particularly user friendly.
```log
2 validation errors for ParsingModel[Dict[str, synapse.config.workers.InstanceLocationConfig]]
__root__ -> event_persister1 -> extra_stuff
  extra fields not permitted (type=value_error.extra)
__root__ -> event_persister2 -> extra_stuff
  extra fields not permitted (type=value_error.extra)
instance_map{'event_persister1': {'extra_stuff': 'something', 'host': 'localhost', 'port': 18009}, 'event_persister2': {'extra_stuff': 'something', 'host': 'localhost', 'port': 18010}, 'stream_writers1': {'host': 'localhost', 'port': 18023}}
```

So I would have to agree that 'ignore' seems, for now, to be the simplest path forward.